wsrep_gtid.c

```diff
@@ -59,13 +61,15 @@ wsrep_gtid_scan(const char* str, size_t str_len, wsrep_gtid_t* gtid)
 int
 wsrep_gtid_print(const wsrep_gtid_t* gtid, char* str, size_t str_len)
 {
-    unsigned int offset, ret;
-    if ((offset = wsrep_uuid_print(&gtid->uuid, str, str_len)) > 0)
+    size_t offset;
+    int ret;
+    if ((ret = wsrep_uuid_print(&gtid->uuid, str, str_len)) > 0)
     {
+        offset = (size_t)ret;
         ret = snprintf(str + offset, str_len - offset,
                        ":%" PRId64, gtid->seqno);
-        if (ret <= str_len - offset) {
-            return (offset + ret);
+        if (ret <= (int)(str_len - offset)) {
+            return ((int)offset + ret);
```

Comment:
it feels like in this function there would be much less mess and casting if `offset` were declared `int`